samples/wifi_provisioning_over_ble/src/main.c

```diff
@@ -167,28 +167,33 @@ void do_wifi_disconnect(void)
  **/
 void do_wifi_connect(void)
 {
-	struct wifi_connect_req_params *cnx_params = NULL;
-
-	cnx_params = malloc(sizeof(struct wifi_connect_req_params));
-	if (cnx_params == NULL) {
-		printk("\r\nmalloc failed\r\n");
-		return;
+	struct wifi_connect_req_params cnx_params = {
+		.channel = WIFI_CHANNEL_ANY,
+		.band = 0,
+		.security = sec_type,
+		.ssid = ssid,
+		.ssid_length = strlen(ssid),
+		.psk = pwd,
+		.psk_length = strlen(pwd),
+	};
+	/* Configure MFP/credentials for WPA3 and mixed modes */
+	cnx_params.mfp = WIFI_MFP_DISABLE;
+	if (sec_type == WIFI_SECURITY_TYPE_SAE_AUTO) {
+		cnx_params.mfp = WIFI_MFP_REQUIRED;
+		cnx_params.sae_password = pwd;
+		cnx_params.sae_password_length = strlen(pwd);
+		cnx_params.psk = NULL;
+		cnx_params.psk_length = 0;
+	} else if (sec_type == WIFI_SECURITY_TYPE_WPA_AUTO_PERSONAL) {
+		cnx_params.mfp = WIFI_MFP_OPTIONAL;
+		cnx_params.psk = pwd;
+		cnx_params.psk_length = strlen(pwd);
```

Comment:
You initialize `psk` during declaration, but you overwrite this value in all cases. I suggest:
```
	struct wifi_connect_req_params cnx_params = {
		.channel = WIFI_CHANNEL_ANY,
		.band = 0,
		.security = sec_type,
		.mfp = WIFI_MFP_DISABLE,
		.ssid = ssid,
		.ssid_length = strlen(ssid),
	};
	if (sec_type == WIFI_SECURITY_TYPE_SAE_AUTO) {
		cnx_params.mfp = WIFI_MFP_REQUIRED;
		cnx_params.sae_password = pwd;
		cnx_params.sae_password_length = strlen(pwd);
	} else if (sec_type == WIFI_SECURITY_TYPE_WPA_AUTO_PERSONAL) {
		cnx_params.mfp = WIFI_MFP_OPTIONAL;
		cnx_params.psk = pwd;
		cnx_params.psk_length = strlen(pwd);
	}
```

Reply:
Indeed. Can you apply this suggestion?